src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,96 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    public static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private Object[] elements;
+
+    public ArrayList() {
+        elements = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        grow();
```

Comment:
Remove length checking from grow() method. Grow() should be responsible for growing but not for checking size.
Otherwise it should be named like "checkLengthAndGrow()"
```suggestion
if (elements.length == size){
    grow();
}        
```